### PR TITLE
Add blog post about Micronaut Test Resources

### PR DESCRIPTION
**Pull request by melix**

### CODE REVIEW

**Comment by melix** [comment]
Screenshot 2022-07-27 at 16-27-24 Cédric Champeau's blog Introducing Micronaut Test Resources

**Comment by melix** [comment]
Link to release notes will have to be updated to the actual 3.6 release

File: src/jbake/content/2022/07/micronaut-test-resources.adoc [hunk 1]
```diff
@@ -0,0 +1,568 @@
...
+The new https://micronaut.io/2022/05/26/micronaut-framework-3-5-0/[release of Micronaut 3.6] introduces a new feature which I worked on for the past couple of months, called {mtr}[Micronaut Test Resources].
+This feature, which is inspired from https://quarkus.io/guides/dev-services[Quarkus' Dev Services], will greatly simplify testing of Micronaut applications, both on the JVM and using GraalVM native images.
+Let's see how.
+
+== Test resources in a nutshell
+
+{mtr}[Micronaut Test Resources] simplifies testing of applications which depend on external resources, by handling the provisioning and lifecycle of such resources automatically.
+For example, if your application requires a MySQL server, in order to test the application, you need a MySQL database to be installed and configured, which includes a database name, a username and a password.
+In general, those are only relevant for production, where those are fixed.
+During development, all you care about is to have _one_ database available.
```
Comment (timyates) [inline src/jbake/content/2022/07/micronaut-test-resources.adoc:21]:
```suggestion
During development, all you care about is having _one_ database available.
```

File: src/jbake/content/2022/07/micronaut-test-resources.adoc [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+
+=== Keeping the service alive
+
+By default, the test resources service is _short lived_: it's going to be started at the beginning of a build, and shutdown at the end of a build.
+This means, that it will live as long as you have tests running, or, if running in development mode, as long as the application is alive.
+However, you can make it survive the build, and reuse the containers in several, independent build invocations.
+
+To do this, you need to _explicitly start the test resources service_:
+
+`./gradlew startTestResourcesService`
```
Comment (timyates) [inline src/jbake/content/2022/07/micronaut-test-resources.adoc:311]: Is there a maven analog of this?

File: src/jbake/content/2022/07/micronaut-test-resources.adoc [hunk 3]
```diff
@@ -0,0 +1,568 @@
...
+    }
+]
+----
+
+Nice, right? 
+However there's a gotcha if you do this: what happens if we run tests?
+
+[source,bash]
+----
+$ gw test
```
Comment (timyates) [inline src/jbake/content/2022/07/micronaut-test-resources.adoc:374]:
```suggestion
$ ./gradlew test
```

**Review by timyates — APPROVED** [review]
Awesome 😎 

Left a couple of suggestions, which can be ignored at will 😋

File: src/jbake/content/2022/07/micronaut-test-resources.adoc (hunk 2 above)
Comment (melix) [reply src/jbake/content/2022/07/micronaut-test-resources.adoc:311]: Yes, I will add it :+1:

**Comment by wetted** [comment]
You might want to mention that Micronaut Guides are getting migrated to use test resources

File: src/jbake/content/2022/08/micronaut-test-resources.adoc [hunk 4]
```diff
@@ -0,0 +1,568 @@
...
+
+In this blog post, we've explored the new {mtr}[Micronaut Test Resources] module, which will greatly simplify development of Micronaut applications which depend on external services like databases or messaging queues.
+It works by _simplifying_ configuration, by removing lines which used to be present, but now are dynamically resolved, like `datasources.default.url`.
+Test resources are handled in a separate process, the test resources server, which is responsible for handling their lifecycle.
+This also makes it possible to share the resources (containers, databases, ...) between independent builds.
+For advanced use cases, {mtr}[Micronaut Test Resources] provides _configuration based_ resources creation.
+
+Last but not least, {mtr}[Micronaut Test Resources] is an extensible framework which will let you implement your own test resources in case the built-in ones miss a feature.
+
+Special thanks to twitter.com/tim_yates/[Tim Yates] for his hard work on upgrading the Micronaut Guides to use test resources, and https://twitter.com/alvaro_sanchez[Álvaro Sanchez-Mariscal] for his support on the Maven plugin!
```
Comment (timyates) [inline src/jbake/content/2022/08/micronaut-test-resources.adoc:567]: Runs in late...

Missing `https://` to make me a link. Thanks for the mention btw 😎

Comment (melix) [reply src/jbake/content/2022/08/micronaut-test-resources.adoc:567]: Fixed, thanks!

**Pull request merged**